app/views/idv/doc_auth/verify.html.erb

```diff
@@ -1,5 +1,33 @@
 <div id="form-steps-wait-alert"></div>
 
+<% if flow_session[:had_barcode_read_failure] %>
+  <%= render AlertComponent.new(
+        type: :warning,
+        class: 'margin-bottom-4',
+        text_tag: 'div',
+      ) do %>
+    <%= t(
+          'doc_auth.headings.capture_scan_warning_html',
+          link: link_to(
+            t('doc_auth.headings.capture_scan_warning_link'),
+            idv_doc_auth_step_path(step: :document_capture),
+            id: 'redo_doc_auth_link',
+          ),
+        ) %>
+    <%= form_tag(
+          idv_doc_auth_step_path(step: :redo_doc_auth),
+          method: :put,
+          id: 'redo_doc_auth_form',
+        ) %>
+    <%= javascript_tag(nonce: true) do %>
+      document.getElementById('redo_doc_auth_link').addEventListener('click', function(event) {
+        event.preventDefault();
+        document.getElementById('redo_doc_auth_form').submit();
+      });
+    <% end %>
```

Comment:
This is an unfortunate bit of code that I added. Normally we would (and did, prior to f3ce40689d6c7120ba09cbbdaa81567f4054e8c4) just use a submit button element and style it to look like a link. However, this negatively impacts text wrapping, and from what I can tell, there is no way to otherwise force a button element to wrap as though it is inline content.

Before|After
---|---
![Screen Shot 2022-06-22 at 9 57 46 AM](https://user-images.githubusercontent.com/1779930/175049411-742370df-b98b-4aef-b658-06f236fd06f8.png)|![Screen Shot 2022-06-22 at 9 57 13 AM](https://user-images.githubusercontent.com/1779930/175049424-8d2c3abb-e3d0-449f-959d-165fb21d888e.png)

This is basically the same thing as what [Rails' "unobtrusive JavaScript" adapter](https://github.com/rails/rails/tree/main/actionview/app/assets/javascripts) does for links to submit as non-GET requests. It might make sense to either pull that in, or at least extract this to a JavaScript pack (maybe make it part of ButtonComponent and have it be reusable?).

Reply:
Refactored to new button behavior (57545ca, 2300e96) in 8f856f4.